spec/cache/asset_spec.rb

```diff
@@ -61,16 +61,30 @@
   end
 
   describe '#unassign_asset' do
-    it 'unassigns the asset from the node' do
-      cache.unassign_asset('asset_test')
+    let :asset_name { 'asset_test' }
+    before :each do
+      cache.assign_asset_to_node(asset_name, node)
+      cache.save
+    end
+
+    it 'unassigns the asset from all found instances ' do
+      cache.unassign_asset(asset_name)
+      cache.save
+      new_cache = Metalware::Cache::Asset.new
+      expect(new_cache.data).not_to eq(content)
```

Comment:
Using `not_to eq` in this context isn't the best way to test `unassign_asset`. All it tests is that the cache has changed from what ever is contained in `content`.

Side note, `content` should be renamed `initial_content` as the "content" is changing and thus `content` could be multiple different "contents".

These spec do not test if `unassign_asset` is working correctly, as it could have done any number of alterations to the cache and the spec will still pass.

The purpose of the spec is to document the expected behaviour. Currently from the spec only, I can not tell what it is suppose to do.

Possible something along these lines:
```
expected_content = initial_content.dup.tap do |c|
  c.delete(node_name)
end
expect(new_cache.data).to eq(expected_content)
```

Designing the spec likes this will require you to know which node will be delete from the cache in advance. However this is kinda the point of the spec to show how the `unassign` will work.

A similar comment applies for the remaining specs. Consider how similar they are and try and extract the setup as much as possible.